Runtime/Scripts/BaseMeshSync.Modifiers.cs

```diff
@@ -0,0 +1,544 @@
+        {
+            StringBuilder sb = new StringBuilder(prefix);
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (i < array.Length)
+                {
+                    sb.Append("|");
+                }
+
+                sb.Append(array[i]);
+            }
+
+            return sb.ToString();
+        }
+
+        public string GetSerializedValue(bool useNewValues = false)
+        {
+            var valueToSerialize = propertyValue;
+
+            if (useNewValues && newValue != null)
+            {
+                valueToSerialize = newValue;
+            }
+
+            if (valueToSerialize == null)
+                return "n";
+            else if (valueToSerialize is int)
+                return "i" + valueToSerialize.ToString();
+            else if (valueToSerialize is float)
+                return "f" + valueToSerialize.ToString();
+            else if (valueToSerialize is string)
+                return "s" + valueToSerialize.ToString();
+            else if (valueToSerialize is int[] intArray)
+                return SaveArray("a", intArray);
+            else if (valueToSerialize is float[] floatArray)
+                return SaveArray("b", floatArray);
+            else
+                throw new NotImplementedException($"propertyValue: {valueToSerialize.GetType()} cannot be serialized!");
+        }
+
+        static object DeserializeString(string serializedValue)
+        {
+            if (serializedValue.Length == 0)
+                return null;
+            char type = serializedValue[0];
+            if (type == 'n')
+                return null;
+            else if (type == 'i')
+                return int.Parse(serializedValue.Substring(1));
+            else if (type == 'f')
+                return float.Parse(serializedValue.Substring(1));
+            else if (type == 's')
+                return serializedValue.Substring(1);
+            else if (type == 'a')
+                return ParseArray(serializedValue, int.Parse);
+            else if (type == 'b')
+                return ParseArray(serializedValue, float.Parse);
+            else
+                throw new NotImplementedException($"propertyValue: {type} cannot be deserialized!");
+        }
+
+        public void SetSerializedValue(string serializedValue)
+        {
+            NewValue = DeserializeString(serializedValue);
+        }
+
+        public void OnBeforeSerialize()
+        {
+            propertyValueSerialized = GetSerializedValue();
+        }
+
+        public void OnAfterDeserialize()
+        {
+            propertyValue = DeserializeString(propertyValueSerialized);
+        }
+
+        public object NewValue
+        {
+            get => newValue;
+            set
+            {
+                switch (type)
+                {
+                    case PropertyInfoDataType.IntArray:
+                        {
+                            var newValueAsArray = new int[arrayLength];
```

Comment:
I tried that but the casting causes issues because I cannot enforce that the type is an int or float and Math.Clamp only takes those. I tried passing in the clamp function but then it doesn't compile because the floats from the vectors cannot cast to T.